### PR TITLE
Fix #3574 add driftpixel userSync redirect url

### LINKED ISSUE #3574: Port PR from PBS-Go: Driftpixel: Add default cookie sync endpoint
A PR was merged over on PBS-Go

- [https://github.com/prebid/prebid-server/pull/3973](https://github.com/prebid/prebid-server/pull/3973)
- timestamp: 2024-11-22T18:01:56Z

### PR DESCRIPTION
**Pull request by muuki88**
### 🔧 Type of changes

- [x] configuration

### ✨ What's the context?

Ported from Go version: https://github.com/prebid/prebid-server/pull/3973

### 🧠 Rationale behind the change

Keep bidder implementations in sync

### 🧪 Test plan

:crossed_fingers: 

### 🏎 Quality check
- [x] Are your changes following [our code style guidelines](https://github.com/prebid/prebid-server-java/blob/master/docs/developers/code-style.md)?
- [ ] Are there any breaking changes in your code?
- [ ] Does your test coverage exceed 90%?
- [ ] Are there any erroneous console logs, debuggers or leftover code in your changes?

### CODE REVIEW

**Comment by muuki88** [comment]
Seems the currency service failed

```
2024-12-11T09:10:21.279Z  WARN 2065 --- [           main] o.p.s.c.CurrencyConversionService        : Error occurred while request to currency service

org.prebid.server.exception.PreBidException: HTTP status code 503
	at org.prebid.server.currency.CurrencyConversionService.processResponse(CurrencyConversionService.java:108)
	at io.vertx.core.impl.future.Mapping.onSuccess(Mapping.java:35)
````

**Comment by osulzhenko** [comment]
@muuki88 We are not using external services in the current tests. Also, please update your redirect URL. It has some differences compared to Go version. You can refer to other bidders with similar configurations. For example, [Vidoomy](https://github.com/prebid/prebid-server-java/blob/master/src/main/resources/bidder-config/vidoomy.yaml)

**Comment by muuki88** [comment]
Done

**Comment by muuki88** [comment]
I'm confused why the tests fail :see_no_evil:

**Comment by marki1an** [comment]
@muuki88 Hi, I found an issue that looks like:
`Property: adapters.driftpixel.usersync.redirect.supportCors
    Value: "null"
    Reason: must not be null
`

**Comment by muuki88** [comment]
Thanks a lot @marki1an 

I'll take a look. Should I research for a way to make the test output more accessible? There were quite a lot of exceptions that all seemed unrelated

**Comment by muuki88** [comment]
```
APPLICATION FAILED TO START
***************************

Description:

Binding to target org.springframework.boot.context.properties.bind.BindException: Failed to bind properties under 'adapters.driftpixel' to org.prebid.server.spring.config.bidder.model.BidderConfigurationProperties failed:

    Property: adapters.driftpixel.usersync.redirect.supportCors
    Value: "null"
    Reason: must not be null


Action:

Update your application's configuration
```

Okay. This crashed the entire application start :see_no_evil:

**Comment by muuki88** [comment]
I'm not 100% sure that there are no external services involved :sweat_smile: 

```
org.prebid.server.hooks.modules.greenbids.real.time.data.v1.GreenbidsRealTimeDataProcessedAuctionRequestHookTest.callShouldFilterBiddersBasedOnModelWhenAnyFeatureNotAvailable -- Time elapsed: 2.842 s <<< ERROR!
java.io.FileNotFoundException: https://github.com/P3TERX/GeoLite.mmdb/releases/download/2024.12.13/GeoLite2-Country.mmdb
```

File: src/main/resources/bidder-config/driftpixel.yaml [hunk 1]
```diff
@@ -13,3 +13,9 @@ adapters:
         - native
       supported-vendors:
       vendor-id: 0
+    userSync:
+      cookie-family-name: driftpixel
+      redirect:
+        url: "https://sync.driftpixel.live/psync?t=s&e=0&gdpr={{gdpr}}&consent={{gdpr_consent}}&us_privacy={{us_privacy}}&cb={{redirect_url}}"
+        support-cors: false
+        userMacro: "%USER_ID%"
```
Comment (AntoxaAntoxic) [inline src/main/resources/bidder-config/driftpixel.yaml:21]:
```yaml
    usersync:
      cookie-family-name: driftpixel
      redirect:
        url: "https://sync.driftpixel.live/psync?t=s&e=0&gdpr={{gdpr}}&consent={{gdpr_consent}}&us_privacy={{us_privacy}}&cb={{redirect_url}}"
        support-cors: false
        uid-macro: '%USER_ID%'
```

**Review by AntoxaAntoxic — CHANGES_REQUESTED** [review]

**Comment by muuki88** [comment]
Thanks @AntoxaAntoxic for your patience :pray: 

I think there's still a problem unrelated to this change. I opened an issues #3629

**Comment by AntoxaAntoxic** [comment]
Please merge the latest master

**Comment by AntoxaAntoxic** [comment]
@muuki88 Not sure I have the rights to do it. @Net-burst can you assist on that?

P.S. The failed test is not related to your changes, so please ignore.

**Review by AntoxaAntoxic — APPROVED** [review]

**Review by CTMBNara — APPROVED** [review]

**Pull request merged**